### PR TITLE
Enhance token management and add Disconnect-AzResourceGraph function

### PR DESCRIPTION
**Pull request by SimonWahlin**
Improve token handling by refining parameter management and adding a new function to disconnect from Azure Resource Graph, clearing cached tokens.

### CODE REVIEW

File: AzResourceGraph/Private/Test-AzureToken.ps1 [hunk 1]
```diff
@@ -46,7 +46,12 @@ function Test-AzureToken {
     )
     return (
         $null -ne $Token -and
-        $Token.ExpiresOn -ge [System.DateTimeOffset]::Now.AddMinutes($MinValid) -and
-        $Token.Claims['aud'] -eq $Resource
+        $Token.ExpiresOn.UtcDateTime -ge [System.DateTimeOffset]::Now.AddMinutes($MinValid).UtcDateTime -and
```
[the remark this answers is not in the document]
Comment (SimonWahlin) [reply AzResourceGraph/Private/Test-AzureToken.ps1:49]: That was how it was done before but I had a problems with edge cases where the offset was invalid but the UtcDateTime was correct so to be on the safe side I chose to compare the UtcDateTime.
